File: include/PolyWorksSDK/samples/plugins/common/CSharp/CSharpCommands/COM_SDK/Util/CComCommandParameterEnum.cs
// =================================================================================================
// Class for COM command parameters enumeration
//
// Copyright © InnovMetric Logiciels Inc. 2009 All Rights Reserved
// -------------------------------------------------------------------------------------------------
namespace IM
{
    [System.Runtime.InteropServices.ComVisible(true)]
    [System.Runtime.InteropServices.ClassInterface(System.Runtime.InteropServices.ClassInterfaceType.None)]
    public class CComCommandParameterEnum : IMPluginLib.IIMCommandParameterEnum
    {
        /* *****************************************************************************
        *                            IIMCommandParameterEnum Interface
        * -------------------------------------------------------------------------- */
        #region IIMCommandParameterEnum

        public int ParamsGetNb()
        {
            if(m_impl != null)
            {
                return m_impl.ParamsGetNb();
            }
            return 0;
        }

        public void ParamGet(int number_, out IMPluginLib.IIMCommandParameter iIMCommandParameter_)
        {
            iIMCommandParameter_ = null;
            if (m_impl != null)
            {
                m_impl.ParamGet(number_, out iIMCommandParameter_);
            }
        }

        #endregion

        /* *****************************************************************************
        *              CComCommandParameterEnum specific methods
        * -------------------------------------------------------------------------- */
        #region CComCommandParameterEnum

        // =============================================================================
        // Constructor
        // -----------------------------------------------------------------------------
        public CComCommandParameterEnum()
        {
            m_impl = new CComCommandParameterEnumImpl();
        }

        // =============================================================================
        // Add an argument to the enumeration
        //
        // Parameter: pCommandParameter_ : Argument to add
        // -----------------------------------------------------------------------------
        public void CommandParameterAdd(IMPluginLib.IIMCommandParameter commandParameter_)
        {
            if (m_impl != null)
            {
                m_impl.CommandParameterAdd(commandParameter_);
            }
        }

        private CComCommandParameterEnumImpl m_impl = null;

        #endregion
    }
}

File: include/PolyWorksSDK/samples/plugins/common/VisualBasic.NET/VBNetCommands/COM_SDK/Util/CComCommandParameterEnum.vb
' =================================================================================================
' Class for COM command parameters enumeration
'
' Copyright © InnovMetric Logiciels Inc. 2011 All Rights Reserved
' -------------------------------------------------------------------------------------------------
Namespace VBNetCommands

    <System.Runtime.InteropServices.ComVisible(True)> _
    <System.Runtime.InteropServices.ClassInterface(System.Runtime.InteropServices.ClassInterfaceType.None)> _
    Public Class CComCommandParameterEnum
        Implements IMPluginLib.IIMCommandParameterEnum

        ' =============================================================================
        '                            IIMCommandParameterEnum Interface
        ' =============================================================================

        Public Function ParamsGetNb() As Integer _
        Implements IMPluginLib.IIMCommandParameterEnum.ParamsGetNb

            If Not m_impl Is Nothing Then

                Return m_impl.ParamsGetNb()

            End If

            Return 0

        End Function

        Public Sub ParamGet(ByVal number_ As Integer, ByRef iIMCommandParameter_ As IMPluginLib.IIMCommandParameter) _
        Implements IMPluginLib.IIMCommandParameterEnum.ParamGet

            iIMCommandParameter_ = Nothing

            If Not m_impl Is Nothing Then

                m_impl.ParamGet(number_, iIMCommandParameter_)

            End If

        End Sub

        ' =============================================================================
        '              CComCommandParameterEnum specific methods
        ' =============================================================================

        ' =============================================================================
        ' Constructor
        ' -----------------------------------------------------------------------------
        Public Sub New()

            m_impl = New CComCommandParameterEnumImpl()

        End Sub

        ' =============================================================================
        ' Add an argument to the enumeration
        '
        ' Parameter: pCommandParameter_ : Argument to add
        ' -----------------------------------------------------------------------------
        Public Sub CommandParameterAdd(ByVal commandParameter_ As IMPluginLib.IIMCommandParameter)

            If Not m_impl Is Nothing Then

                m_impl.CommandParameterAdd(commandParameter_)

            End If

        End Sub

        Private m_impl As CComCommandParameterEnumImpl = Nothing

    End Class

End Namespace



